spec/requests/projects_spec.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ProjectsController, type: :request do
+
+  describe 'GET /projects/:id/use_as_template' do
+    let!(:submitter) { create :user }
+    let!(:season)    { nil }
+    let!(:project)   { create :project, season: season, submitter: submitter }
+
+    shared_examples_for 'returns to the root page' do
+      it 'returns to the root page' do
+        get use_as_template_project_path(project), headers: { 'HTTP_REFERER' => '/previouspage' }
+        expect(response).to redirect_to '/' # load_and_authorize_resource ignores the referrer
+        expect(flash[:alert]).to match %r{not authorized}
+      end
+    end
+
+    context 'when the user is not logged in' do
+      it_behaves_like 'returns to the root page'
+    end
+
+    context 'when the user is logged in' do
+      let!(:user) { create :user }
```

Comment:
No need for the bang here, the `before` hook will make sure the user is created.